src/posix/main.c

```diff
@@ -153,6 +154,7 @@ static const struct option kOptions[] = {
     {"real-time-signal", required_argument, NULL, OT_POSIX_OPT_REAL_TIME_SIGNAL},
     {"time-speed", required_argument, NULL, OT_POSIX_OPT_TIME_SPEED},
     {"verbose", no_argument, NULL, OT_POSIX_OPT_VERBOSE},
+    {"settings-path", required_argument, NULL, OT_POSIX_OPT_SETTINGS_PATH},
```

Comment:
nit: since this is not only for settings, but also for tmp storage, perhaps rename to data-path?
```suggestion
    {"data-path", required_argument, NULL, OT_POSIX_OPT_DATA_PATH},
```